.cargo/config.toml

```diff
@@ -0,0 +1,2 @@
+[env]
+JEMALLOC_SYS_WITH_MALLOC_CONF = "percpu_arena:percpu,oversize_threshold:0,muzzy_decay_ms:5000,dirty_decay_ms:5000"
```

Comment:
How about add a link to https://github.com/jemalloc/jemalloc/blob/dev/TUNING.md ?

Reply:
afraid that the configuration used in this PR is not derived from the official tuning guidance. the essential option `oversize_threshold:0` is not mentioned in TUNING.md

Instead, it is from clickhouse :

https://github.com/ClickHouse/ClickHouse/blob/533c7888038453c047df816f3f65292cca05a54f/contrib/jemalloc-cmake/CMakeLists.txt#L37

we have evaluated this configuration before, but no significant improvement been observed. 

But, after @zhang2014 's PR https://github.com/datafuselabs/databend/pull/9379 has been merged, the above malloc conf finally takes effect.

Let me add the above info the PR's summary